src/java/org/openzal/zal/MailboxData.java

```diff
@@ -31,12 +68,43 @@ public int getSchemaGroupId()
     return mSchemaGroupId;
   }
 
+  @Nullable
   public String getAccountId()
   {
     return mAccountId;
   }
 
-  public short getIndexVolumeId()
+  @Override
+  public boolean equals(Object o)
+  {
+    if( this == o )
+    {
+      return true;
+    }
+    if( o == null || getClass() != o.getClass() )
+    {
+      return false;
+    }
+    MailboxData that = (MailboxData) o;
+    if(mId != null && that.mId != null && !Objects.equals(mId, that.mId))
+    {
+      return false;
+    }
+    if(mAccountId != null && that.mAccountId != null && !Objects.equals(mAccountId, that.mAccountId))
+    {
+      return false;
+    }
```

Comment:
Other fields as the volumeId are to not take in account during equals?

Reply:
MailboxDatas are equals whether mailboxId/accountId are equals (and only if are both present).